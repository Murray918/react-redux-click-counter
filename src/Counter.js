import React from 'react';
import { connect } from 'react-redux';
//this is a specific react-redux function called connect



//pure functional component

function Counter (props) {
  console.log('renderrrrr', props);
  return (
    <div>
      <h1>We be countin' up in here</h1>
      <p> Count: {props.count}</p>
      <button> Click Counter </button>
    </div>
  )
}

export default Counter;
