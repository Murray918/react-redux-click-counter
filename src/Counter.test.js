import React from 'react';
import Counter from './Counter.js';
import renderer from 'react-test-renderer';


test('Counter renders properly', () => {
  const tree = renderer.create(
    <h1>We be countin' up in here</h1>
  ).toJSON();
    expect(tree).toMatchSnapshot();
});
