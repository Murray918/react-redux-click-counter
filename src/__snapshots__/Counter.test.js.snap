// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders properly 1`] = `
<h1>
  We be countin' up in here
</h1>
`;
